pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter_test.go

```diff
@@ -4332,6 +4332,202 @@ var _ = Describe("Validating VMICreate Admitter", func() {
 		})
 
 	})
+
+	Context("with topologySpreadConstraints checks", func() {
+		var vmi *v1.VirtualMachineInstance
+		BeforeEach(func() {
+			vmi = api.NewMinimalVMI("testvmi")
+			vmi.Spec.TopologySpreadConstraints = []k8sv1.TopologySpreadConstraint{}
+		})
+		It("Allow to create when spec.topologySpreadConstraints set to nil", func() {
+			vmi.Spec.TopologySpreadConstraints = nil
+			vmiBytes, _ := json.Marshal(&vmi)
+
+			ar := &admissionv1.AdmissionReview{
+				Request: &admissionv1.AdmissionRequest{
+					Resource: webhooks.VirtualMachineInstanceGroupVersionResource,
+					Object: runtime.RawExtension{
+						Raw: vmiBytes,
+					},
+				},
+			}
+
+			resp := vmiCreateAdmitter.Admit(ar)
```

Comment:
Would it make sense to create a function that sets the given `TopologySpreadConstraints` for the `vmi`? The above code seems to be used in all the tests. This will reduce duplication.

Reply:
I think so, will update

Reply:
@vasiliy-ul I see that a lot of code duplication is happening due to 
```
			ar := &admissionv1.AdmissionReview{
				Request: &admissionv1.AdmissionRequest{
					Resource: webhooks.VirtualMachineInstanceGroupVersionResource,
					Object: runtime.RawExtension{
						Raw: vmiBytes,
					},
				},
			}

			resp := vmiCreateAdmitter.Admit(ar)
```

And that is across all the tests.  Since the scope of that is larger, can I take that up in a follow up cleanup PR? or may be address in https://github.com/kubevirt/kubevirt/pull/8239

Reply:
Still, I would consider creating a local helper function within the topologySpreadConstraints context scope with the following content:

```go
			vmi.Spec.TopologySpreadConstraints = topologySpreadConstraints
			vmiBytes, _ := json.Marshal(&vmi)

			ar := &admissionv1.AdmissionReview{
				Request: &admissionv1.AdmissionRequest{
					Resource: webhooks.VirtualMachineInstanceGroupVersionResource,
					Object: runtime.RawExtension{
						Raw: vmiBytes,
					},
				},
			}

			return vmiCreateAdmitter.Admit(ar)
```
This will not extend the current scope.

But I do not insist. If you prefer to do that outside of this PR I am okay with that.